packages/kubernetes/0005-opt-out-of-module-mode-for-builds.patch

```diff
@@ -78,331 +54,20 @@ index 42f34567db9..dffbc01e113 100755
 +export GO111MODULE=off
  
  # The root of the build/dist directory
--KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE}")/../.." && pwd -P)"
-+KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
- 
- KUBE_OUTPUT_SUBPATH="${KUBE_OUTPUT_SUBPATH:-_output/local}"
- KUBE_OUTPUT="${KUBE_ROOT}/${KUBE_OUTPUT_SUBPATH}"
-@@ -44,7 +44,7 @@ KUBE_RSYNC_COMPRESS="${KUBE_RSYNC_COMPRESS:-0}"
- export no_proxy=127.0.0.1,localhost
- 
- # This is a symlink to binaries for "this platform", e.g. build tools.
--THIS_PLATFORM_BIN="${KUBE_ROOT}/_output/bin"
-+export THIS_PLATFORM_BIN="${KUBE_ROOT}/_output/bin"
- 
- source "${KUBE_ROOT}/hack/lib/util.sh"
- source "${KUBE_ROOT}/hack/lib/logging.sh"
-@@ -56,13 +56,16 @@ source "${KUBE_ROOT}/hack/lib/golang.sh"
- source "${KUBE_ROOT}/hack/lib/etcd.sh"
- 
- KUBE_OUTPUT_HOSTBIN="${KUBE_OUTPUT_BINPATH}/$(kube::util::host_platform)"
-+export KUBE_OUTPUT_HOSTBIN
- 
- # list of all available group versions.  This should be used when generated code
- # or when starting an API server that you want to have everything.
- # most preferred version for a group should appear first
- KUBE_AVAILABLE_GROUP_VERSIONS="${KUBE_AVAILABLE_GROUP_VERSIONS:-\
- v1 \
-+admissionregistration.k8s.io/v1 \
```

Comment:
I'm not terribly familiar with the Kubernetes build process, but this looks like it affects the user-facing API.  Can you confirm that these parameters *don't* change the behavior of the Kubelet?

Reply:
The patch only affected build/test code, but still, we didn't want/intend to affect the API within the build/test process.  This fixes that.

Reply:
I don't understand your reply; maybe my question wasn't clear enough?

We already have a patch in Bottlerocket which disables the module-based build, and you're simplifying that patch.  One of the parts of the patch you're removing is this set of lines, that seems to be related to the features enabled in the Kubernetes API.  I believe this configuration only applies to the Kubernetes API server, which we don't package in Bottlerocket.  However, I want to make sure that this belief is correct; I don't want us to unintentionally remove functionality that we've already included in Bottlerocket by accidentally making a change that affects the Kubelet's behavior.

Reply:
The original patch intended to include just the commit that fixed the build with respect to go modules, but accidentally included a number of other commits as well - things related to shellcheck cleanup, test output format, and yes, API version support.  We didn't want those changes in our patch, we just wanted the build to work with respect to modules.  We're not affecting functionality already included in Bottlerocket because the additional commits that were accidentally pulled in only affect the build/test process, not runtime.  I looked through any usage of the variable you pointed out above (the list of API support) and at the codegen, and do not believe it impacts anything we use; as you mentioned, we don't use the API server, nor do we use generated clients or other things it seems to generate.  If you think we do use something affected by that generation please let me know.

Reply:
Thank you!  That addresses my concerns.